packages/server/src/integrations/base/sql.ts

```diff
@@ -415,9 +415,7 @@ class InternalBuilder {
     if (opts.disableReturning) {
       return query.insert(parsedBody)
     } else {
-      return query
-        .insert(parsedBody)
-        .returning(generateSelectStatement(json, knex))
```

Comment:
Is this safe? I remember there being a valid reason for using a select statement rather than the total `*` - in theory there will never be any joined data here so we can't have column overlap, but selecting the columns the same way we do when we are actually fetching data could be safer.

Reply:
This is actually the change required to fix the bug. It's just reverting the code to what was previously there: https://github.com/Budibase/budibase/commit/14beca3dde33114a0c87876229596bb5e60f5e56#diff-4c4ab33e19e10728e2cad3fa664f4f4af0b02e2c5bae0eb6eecc4d1421edc13c

Reply:
Interesting this caused the issue - what is the exact problem occurring here?

Reply:
We changed this because this was causing a small bug, returning unexpected relationships in returns when the relationship was null and the both primary column name of the foreign table was being used for the primary table as well. It was a small bug that affected the return of `patches` and `inserts`, probably never affected anybody as it's not common to listen to these responses. But this causes the UI to fail in some cases.
All the changes on this PR are to be able to get it working in both scenarios and getting sure that all type of relationships are supported

Reply:
Thanks for the explanation!